src/main/webapp/app/grading-system/grading-key-overview/grading-key-overview.component.ts

```diff
@@ -0,0 +1,104 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { GradingSystemService } from 'app/grading-system/grading-system.service';
+import { GradeStep } from 'app/entities/grade-step.model';
+import { GradeType } from 'app/entities/grading-scale.model';
+import { round } from 'app/shared/util/utils';
+import { CourseScoreCalculationService, REACHABLE_POINTS } from 'app/overview/course-score-calculation.service';
+
+@Component({
+    selector: 'jhi-grade-key-overview',
+    templateUrl: './grading-key-overview.component.html',
+    styleUrls: ['./grading-key-overview.scss'],
+})
+export class GradingKeyOverviewComponent implements OnInit {
+    constructor(
+        private route: ActivatedRoute,
+        private router: Router,
+        private gradingSystemService: GradingSystemService,
+        private courseCalculationService: CourseScoreCalculationService,
+    ) {}
+
+    isExam = false;
+
+    courseId?: number;
+    examId?: number;
+
+    title?: string;
+    gradeSteps: GradeStep[] = [];
+    studentGrade?: string;
+    isBonus = false;
+
+    ngOnInit(): void {
+        this.route.params.subscribe((params) => {
+            this.courseId = Number(params['courseId']);
+            if (params['examId']) {
+                this.examId = Number(params['examId']);
+                this.isExam = true;
+            }
+            this.gradingSystemService.findGradeSteps(this.courseId, this.examId).subscribe((gradeSteps) => {
```

Comment:
It could work, I guess. I followed what's typically being done in the Artemis code, when there's a subscription regarding the route parameters (basically any major component)